src/main/java/org/nuxeo/ecm/core/storage/marklogic/MarkLogicRepository.java

```diff
@@ -672,13 +682,21 @@ private State findOne(String ctsQuery) {
             logQuery(query);
         }
         // Run query
-        try (Session session = xccContentSource.newSession()) {
-            AdhocQuery request = session.newAdhocQuery(query);
-            // ResultSequence will be closed by Session close
+        try {
+            // Don't auto close session it is the stream responsibility
+            Session session = xccContentSource.newSession();
```

Comment:
But what if there's an exception before the stream is returned? I'd use something like:

    boolean completedAbruptly = true;
    Session session = xccContentSource.newSession()
    try {
        ...
        Stream<State> stream = StreamSupport.stream(...)....; 
        completedAbruptly = false; // the stream takes responsibility for closing the session
        return stream;
    } finally {
        if (completedAbruptly) {
            session.close();
        }
    }